### PR TITLE
Downgrading dependency to solve issue with json4s-core dependency version conflict

### PR DESCRIPTION
**Pull request by pdolega**
Issue described here: https://github.com/sbt/sbt-bintray/issues/104

Before this can be built successfully we need to have first this guy merged: https://github.com/sbt/bintry/pull/28

It fully reproduces the issue (take a look - just change back `"org.foundweekends" %% "bintry" % "0.6.0"`  to `"org.foundweekends" %% "bintry" % "0.5.0"` ).

I do however has one critical problem with running `sbt-scripted` with `sbt 1.0.0-M5` (look at comment on PR)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -31,13 +31,13 @@ lazy val root = (project in file("."))
...
-    sbtVersion in Global := "1.0.0-M5",
-    scalaCompilerBridgeSource :=
-      ("org.scala-sbt" % "compiler-interface" % "0.13.15" % "component").sources
-  )
+//  .settings(
+//    scalaVersion := "2.12.2",
+//    sbtVersion in Global := "1.0.0-M5",
+//    scalaCompilerBridgeSource :=
+//      ("org.scala-sbt" % "compiler-interface" % "0.13.15" % "component").sources
+//  )
```
Comment (pdolega) [inline build.sbt]: Here is the problem - I am not able to deal with the dependencies here:

```
[info] Resolving org.scala-sbt#launcher;1.0.0 ...
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: org.scala-sbt#scripted-sbt;1.0.0-M5: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
```

I got an impression that `sbt-scripted` was published somehow differently for `1.0.0`

Comment (pdolega) [inline build.sbt]: I think that I need to capture somehow the target `sbt` version here. So what I tried in `plugins.sbt` was something like this (doesn't work):

```
def dispatchScriptedName(sbtVersion: String) = {
  println(sbtVersion)
  CrossVersion.sbtApiVersion(sbtVersion) match {
    case Some((1, _)) => "scripted-plugin_2.12"
    case wat => println(wat); "scripted-plugin"
  }
}

libraryDependencies += "org.scala-sbt" % dispatchScriptedName(sbtVersion.value) % sbtVersion.value
```

**Comment by jvican** [comment]
@pdolega This is a great contribution, I'm sorry we didn't chime in before.
  
I'm updating this PR with my own release of bintry 0.6.0 and all your findings to leave it in a state in which it can be merged. It would be good if you chime in and review my changes too.

**Comment by jvican** [comment]
@pdolega Your PR has been rebased to master and I've fixed the scripted failure. Everything should be ok now. Thanks a lot for adding the scripted test.

**Comment by jvican** [comment]
CI is failing because 0.6.0 has not been released for bintry. Can someone with merge rights please do so? I can only publish stuff in sbt-bintray, but not in sbt/bintry.

**Comment by dwijnand** [comment]
Superseded by #123.

**Pull request closed without merging**